bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlProvider.java

```diff
@@ -244,110 +234,106 @@ public boolean hasConfigs() {
     }
 
     @Override
-    public Optional<BArray> getAsArrayAndMark(Module module, VariableKey key) {
+    public Optional<Object> getAsArrayAndMark(Module module, VariableKey key) {
```

Comment:
Api looks inconsistent where some methods we return bvalue and others toml nodes. We can have a generic wrapper class like ConfigValue and implement getValue method. So two impls can be done as CliConfigValue and TomlConfigValue. We can do that change separately as well.

Reply:
Created an issue #31683 for this.